### PR TITLE
Upgrade to latest LSP client implementation and improve typescript configuration

### LINKED ISSUE #348: Switch to ESM & node 18 & newer version of VS Code
latest vscode-language-client (https://github.com/usethesource/rascal-language-servers/pull/321) and chai requires both ESM and node 18. 

Looks like this is something we have to tackle soon.

### PR DESCRIPTION
**Pull request by DavyLandman**
This PR does a few things that we've been delaying:

- finish upgrade to node 18 (more and more of our dependencies stopped working on node 16, and VS Code was now running node 18 as well)
- simplify our typescript config
- upgrade to latest lsp library
- we could drop node-fetch dependency as java 18 has fetch build in.
- switch to esbuild for both faster compilation and less configuration (webpack required quite some extra config)

As a side effect this has to increase our minimal VS Code version to 1.82.0 (which was released in June 2023). Since most users are frequently updating VS Code, this should be fine.

This fixes #348 , we could not switch to ESM, as VS Code does not support that, but we could switch to a newer module system, just so we can use the new lsp library.

### CODE REVIEW

File: rascal-vscode-extension/src/auto-jvm/downloaders.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@
  */
 import * as os from 'os';
 import * as path from 'path';
-import fetch, { Response } from 'node-fetch';
```
Comment (jurgenvinju) [inline rascal-vscode-extension/src/auto-jvm/downloaders.ts:29]: Remove commented code

File: rascal-vscode-extension/src/auto-jvm/downloaders.ts [hunk 2]
```diff
@@ -176,12 +176,23 @@ export async function identifyLatestTemurinLTSRelease(version: number, arch: Tem
         throw new Error("Adoptium returned no releases");
     }
     const rel =releases.versions[0];
+    if (!rel) {
+        throw new Error("Missing versions in api result" + rel);
```
Comment (jurgenvinju) [inline rascal-vscode-extension/src/auto-jvm/downloaders.ts:180]: Cryptic error for user. Include talk about the jdk

**Comment by jurgenvinju** [comment]
Thanks for this upgrade!

File: rascal-vscode-extension/src/auto-jvm/downloaders.ts (hunk 1 above)
Comment (DavyLandman) [reply rascal-vscode-extension/src/auto-jvm/downloaders.ts:29]: oops :) well spotted.

File: rascal-vscode-extension/src/auto-jvm/downloaders.ts (hunk 2 above)
Comment (DavyLandman) [reply rascal-vscode-extension/src/auto-jvm/downloaders.ts]: good point, I've updated it.

**Pull request merged**